extensions/python/PythonScriptEngine.cpp

```diff
@@ -243,4 +243,24 @@ std::vector<core::Relationship> PythonScriptEngine::getCustomPythonRelationships
   return relationships;
 }
 
+void PythonScriptEngine::setModuleAttributes(const std::string& qualified_module_name) {
+  GlobalInterpreterLock gil;
```

Comment:
Reading this, I wonder when exactly we need to hold the GIL, and whether we could make it more foolproof. (for fools like me)

Reply:
From my experience it seems that every time we are calling the python C API we need to be holding the GIL, otherwise one of the python API calls crashes the application with a segmentation fault, so in a way it is foolproof, although debugging the issue is not always trivial as it usually does not crash at call where the GIL was missing.